Client/Frontend/Widgets/AutocompleteTextField.swift

```diff
@@ -0,0 +1,139 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+/// Delegate for the text field events. Since AutocompleteTextField owns the UITextFieldDelegate,
+/// callers must use this instead.
+protocol AutocompleteTextFieldDelegate: class {
+    func autocompleteTextField(autocompleteTextField: AutocompleteTextField, didTextChange text: String)
+    func autocompleteTextFieldShouldReturn(autocompleteTextField: AutocompleteTextField) -> Bool
+    func autocompleteTextFieldShouldClear(autocompleteTextField: AutocompleteTextField) -> Bool
+    func autocompleteTextFieldDidBeginEditing(autocompleteTextField: AutocompleteTextField)
+    func autocompleteTextFieldDidEndEditing(autocompleteTextField: AutocompleteTextField)
+}
+
+/// Data source for the text field autocompletion.
+protocol AutocompleteTextFieldDataSource: class {
+    func autocompleteTextField(autocompleteTextField: AutocompleteTextField, suggestionForText text: String, complete: String? -> Void)
+}
+
+private struct AutocompleteTextFieldUX {
+    static let HighlightColor = UIColor(rgb: 0xccdded)
```

Comment:
I think I agree. Maybe we should have a global Styles struct, perhaps segmented into different sub-structs, and keep them all together there.